### DISCUSSION TITLE
next/image doesn't support `srcSet`?

### DISCUSSION THREAD

**Discussion by kaf-lamed-beyt** [vercel/next.js#43024] [Help] [answered]
I'm trying to have different images rendered on different device sizes but, it doesn't seem to be working whenever I check my page speed insight. 

The "provide responsive image" error still persists.

**Comment by icyJoseph** [comment, marked as answer]
- https://nextjs.org/docs/api-reference/next/image#device-sizes
- https://nextjs.org/docs/api-reference/next/image#image-sizes

> These widths are concatenated with the array of [device sizes](https://nextjs.org/docs/api-reference/next/image#device-sizes) to form the full array of sizes used to generate image [srcset](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/srcset)s.

Could you share the report? Or perhaps the above is enough to guide you?

**Reply by icyJoseph** [reply]
@kaf-lamed-beyt From looking at your example, yeah, `srcSet` is not taken into account. For example if you look at the types, https://github.com/vercel/next.js/blob/canary/packages/next/client/image.tsx#L94, here:

```ts
export type ImageProps = Omit<
  JSX.IntrinsicElements['img'],
  'src' | 'srcSet' | 'ref' | 'alt' | 'width' | 'height' | 'loading'
> & {
  src: string | StaticImport
  alt: string
  width?: SafeNumber
  height?: SafeNumber
  // rest of props
```

So `srcSet` is omitted. You can see here how these are calculated: https://github.com/vercel/next.js/blob/canary/packages/next/client/image.tsx#L232

I guess, you could pass the unoptimized flag to these images, since you already seem to have resized and optimized your images? Not 100% sure though, there's a few implications here that I never got around to test in isolation.

**Reply by kaf-lamed-beyt** [reply]
I read through this file, and I understood the reason behind the issue. 

https://github.com/vercel/next.js/blob/canary/packages/next/client/image.tsx

Thank you for suggesting the `unoptimized` flag

**Comment by kaf-lamed-beyt** [comment]
I have an array of different image sizes in my frontmatter, like so;

```js
imageSet: "
https://res.cloudinary.com/meje/image/upload/f_auto,q_70,w_256/v1668161031/article-covers/i18n_hfp9md.jpg 256w,
  https://res.cloudinary.com/meje/image/upload/f_auto,q_70,w_512/v1668161031/article-covers/i18n_hfp9md.jpg 512w,
  https://res.cloudinary.com/meje/image/upload/f_auto,q_70,w_768/v1668161031/article-covers/i18n_hfp9md.jpg 768w,
  https://res.cloudinary.com/meje/image/upload/f_auto,q_70,w_1024/v1668161031/article-covers/i18n_hfp9md.jpg 1024w,
  https://res.cloudinary.com/meje/image/upload/f_auto,q_70,w_1280/v1668161031/article-covers/i18n_hfp9md.jpg 1280w,
"
```
And I'm using them like this in `next/image` 👇🏼 

```jsx
<Image
  alt={`${data.title} cover`}
  src={data.cover_image}
  priority={true}
  srcSet={data.imageSet}
  sizes="(max-width: 768px) 100vw, (max-width: 1200px), 50vw, 33vw"
/> 
```
Here's the result I'm getting from Pagespeed insights
![Screenshot from 2022-11-17 09-03-44](https://user-images.githubusercontent.com/48127194/202390331-d93585b0-308c-4bcc-94ec-a07a00dcf754.png)
